metal_header/riscv_plic0.c++

```diff
@@ -204,9 +215,26 @@ void riscv_plic0::define_inlines() {
             add_inline_body(funcl, "idx == " + std::to_string(i),
                             std::to_string(irline));
           }
+
+          if (irline == MMODE_EXT_INTR) {
+            uint32_t hartid = m.parent().get_field<uint32_t>("reg");
```

Comment:
This should be an `addr_t` instead of a `uint32_t`, because the CPU in e51-arty.dts has `#address-cells = <2>`. This should also fix the problem that required you to change the `reg` field in e51-arty.dts.

Reply:
Do you mean use ```target_addr```? If so, It is work for me. Thanks a lot.
But it seems to me that we should change the address-cells to 1 in cpus node, because we don't need two cells length to express the value for property of cpu nodes. So far as I know, the dts for 64-bit addresses in Linux, U-boot and freedom-u540-c000-bootloader use the one cells in cpu node. I missed this before, if you also agree that, I would change the address-cells to 1 in cpus node.

Reply:
`target_addr` is right, yeah.

Changing the `#address-cells` in the CPU node works, but in this case I think it's best to leave it be to make sure we can continue to support designs like it.

Reply:
I go along with you. I would change the type to ```target_addr```, and also change the ```address-cells``` in dts.